File: labs/week-05/week05.scala
package main.scala.chapter3

import org.apache.spark.sql.SparkSession
import org.apache.spark.sql.types._

object week05 {
    def main(args : Array[String]){

        val spark = SparkSession
        .builder
        .appName("week-05")
        .getOrCreate
   
    val file = "./Divvy_Trips_2015-Q1.csv"
    val dataFrame = spark.read.csv(file)
    
    dataFrame.show(false)
    println(dataFrame.printSchema)
    println(dataFrame.count())
    
    // schematically

    val dSchema = StructType(Array(StructField("trip_id", IntegerType, false),
    StructField("starttime", StringType, false),
    StructField("stoptime", StringType, false),
    StructField("bikeid", IntegerType, false),
    StructField("tripduration", IntegerType, false),
    StructField("from_station_id", IntegerType, false),
    StructField("from_station_name", StringType, false),
    StructField("to_station_id", IntegerType, false),
    StructField("to_station_name", StringType, false),
    StructField("usertype", StringType, false),
    StructField("gender", StringType, false),
    StructField("birthyear", IntegerType, false)))

    val dDataFrame = spark.read.schema(dSchema)
    .csv(file)

    dDataFrame.show(false)
    println(dDataFrame.printSchema)
    println(dDataFrame.count())

    //DLL

    val ddlSchema = "trip_id INT, starttime STRING, stoptime STRING, BikeID INT, TripDuration INT, from_station_id INT, from_station_name STRING, to_station_id INT, to_station_name STRING, usertype STRING, gender STRING, birthyear INT"

    val DataFrameDdl = spark.read.schema(ddlSchema)
    .csv(file)

    DataFrameDdl.show(false)
    println(DataFrameDdl.printSchema)
    println(DataFrameDdl.count())

    }
}

File: labs/week-05/week-05.py
from pyspark.sql import SparkSession 
from pyspark.sql.types import *

spark = (SparkSession
.builder
.appName("week-05")
.getOrCreate())



dataframe = spark.read.csv("./Divvy_Trips_2015-Q1.csv")
print(dataframe.show())
print(dataframe.printSchema())
print(dataframe.count())

schema = StructType([StructField("trip_id",IntegerType(), False),
    StructField("starttime",StringType(), False),
    StructField("stoptime",StringType(), False),
    StructField("bikeid",IntegerType(), False),
    StructField("tripduration",IntegerType(), False),
    StructField("from_station_id",IntegerType(), False),
    StructField("from_station_name", StringType(), False),
    StructField("to_Station_id",IntegerType(), False),
    StructField("to_Station_name",StringType(), False),
    StructField("usertype",StringType(), False),
    StructField("gender",StringType(), False),
    StructField("birthyear",IntegerType(), False),
    ])

data = spark.read.csv("./Divvy_Trips_2015-Q1.csv",header = True, schema = schema)
print(data.show())
print(data.printSchema())
print(data.count())


# DLL 

ddl_schema = "trip_id INT, starttime STRING, stoptime STRING, BikeID INT, TripDuration INT, from_station_id INT, from_station_name STRING, to_station_id INT, to_station_name STRING, usertype STRING, gender STRING, birthyear INT"

ddl_display = spark.read.csv("./Divvy_Trips_2015-Q1.csv", header = True, schema = ddl_schema)
print(ddl_display.show())
print(ddl_display.printSchema())
print(ddl_display.count())
